Add support for NPM 8, 9, 10 and PNPM 8, remove support for NPM 6 (#7286)
sulu/sulu: UPGRADE.md

Sulu 2.6. This will make it easier for you to identify any bugs that may occur
in the project code with PHP 8.2.

### Custom Admin Builds npm version changed

Sulu 2.6 now supports [npm 8, 9, 10](https://nodejs.org/en/download) and even
[pnpm 8](https://pnpm.io/) or [bun 1](https://bun.sh/) to be used for custom admin

> builds. To support this new versions we were required to drop support for NPM 6.

Author: @Prokyonn can you have a look at this :)

> Sulu 2.6 now supports [npm 8, 9, 10](https://nodejs.org/en/download) and even
> [pnpm 8](https://pnpm.io/) or [bun 1](https://bun.sh/) to be used for custom admin
> builds. To support this new versions we were required to drop support for NPM 6.

Reviewer: ```suggestion
Sulu 2.6 now supports [npm 8, 9, and 10](https://nodejs.org/en/download), 
as well as [pnpm 8](https://pnpm.io/) or [bun 1](https://bun.sh/) for custom 
admin builds. With the introduction of these new versions, it is necessary 
to drop the support for npm 6.
```
